### ISSUE TITLE
Stuck trying to load a save, crash upon trying to cancel

### ISSUE THREAD

**Issue by MorningBeverage** [anegostudios/VintageStory-Issues#2731]
### Game Version

1.18.1

### Platform

Windows

### Modded

None

### SP/MP

Singleplayer

### Description

Crash report said "please copy and paste this into the bug report" so hopefully I'm in the right place. Apologies if this is the wrong place or if something's missing, wrongly formatted, etc.

Game seemed to get stuck trying to load a save, I'd say I waited about 2 minutes before I gave up. It crashed after I clicked "Cancel" and "Force Quit."  Thankfully, next time I opened the game, everything worked smoothly and the world loaded quickly, so I don't know what went wrong before. I figure it can't hurt to report it anyway. Maybe it was something to do with having just turned my computer on? Also, I've never used any mods.
### crash data:
Running on 64 bit Windows with 32 GB RAM 
Game Version: v1.18.1 (Stable)
5/11/2023 12:01:42 PM: Critical error occurred
Loaded Mods: 
System.Exception: Coding error: Mods must not get assets before AssetsLoaded stage - do not load assets in a Start() method!
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadBitmap(ClientMain game, AssetLocationAndSource textureLoc) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 845
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadCompositeBitmap_Single(ClientMain game, AssetLocationAndSource compositeTextureLocation) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 667
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadCompositeBitmap_NoOverlayDispose(ClientMain game, AssetLocationAndSource compositeTextureLocation) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 797
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadBitmapsAsync(BakedBitmap[] bitmaps) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 454
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 5/11/2023 12:01:42 PM, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Exception
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadBitmap(Vintagestory.Client.NoObf.ClientMain, Vintagestory.API.Common.AssetLocationAndSource)
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadCompositeBitmap_Single(Vintagestory.Client.NoObf.ClientMain, Vintagestory.API.Common.AssetLocationAndSource)
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadCompositeBitmap_NoOverlayDispose(Vintagestory.Client.NoObf.ClientMain, Vintagestory.API.Common.AssetLocationAndSource)
   at Vintagestory.Client.NoObf.TextureAtlasManager.LoadBitmapsAsync(Vintagestory.API.Common.BakedBitmap[])
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()

 }
--------------
{ TimeGenerated = 5/2/2023 4:04:53 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1248370421850872525, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.18.1.0
P3: 10.0.19045.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\LATT~1\AppData\Local\Temp\RDRC08F.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERC0A0.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERC0B0.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERC0CE.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERC0EE.tmp.txt

These files may be available here:

Analysis symbol: 
Rechecking for solution: 0
Report Id: cae31a43-ca91-43ee-8ac2-38631024222f
Report Status: 268435456
Hashed bucket: 14d5118b5bf73c7861531a48d6869acd
Cab Guid: 0 }

### How to reproduce

unknown

**Comment by Craluminum2413** [comment]
It looks like you are using mods, are you sure that `\AppData\Roaming\VintagestoryData\Mods` is empty and `\AppData\Roaming\Vintagestory\Mods` has only 6 files?

**Comment by MorningBeverage** [comment]
Checked and there are zero files in `\AppData\Roaming\VintagestoryData\Mods` right now.
`\Vintagestory\Mods` has exactly 6 files, though it's at a different file path.
![vintage story data mods](https://github.com/anegostudios/VintageStory-Issues/assets/47262084/90afc231-6629-4ee3-a48a-6321ab640045)
![vintage story mods](https://github.com/anegostudios/VintageStory-Issues/assets/47262084/71126b48-0c30-442a-8cce-01b1d0ce62fc)
I'm sure I haven't added/removed files in either of those folders before or after the crash, and my world went back to loading normally after the crash.

**Comment by radfast** [comment]
[Similar issue posted by onion on Discord](https://discord.com/channels/302152934249070593/334233095374897152/1105807280680742983)

**Comment by radfast** [comment]
The error seen is because pressing the Quit button caused the assets to be **unloaded**.  But meanwhile, a startup thread was still trying to build a texture atlas for blocks / items / entities.   It could not continue with that, with no assets loaded, so an error is inevitable here even if not well-described.   We can include some shutdown detection code instead for better error reporting, but that won't fix the issue it will just change the error you see.

The issue is why did your game get stuck during startup, at all?   The vanilla game should startup very smooth every time, in 10 to 20 seconds depending on your hardware.

Please start the game again, quit like 30 seconds after it seems to get stuck, and then post up here two logs:  `client-main.txt` and `client-debug.txt`.   Those should give plenty more info to troubleshoot this issue

**Comment by radfast** [comment]
Please also test starting a new world - test both Creative (flat world) and a standard Survival world with default settings.

It may be something strange in your specific savegame is causing this issue.

**Comment by Inbequer** [comment]
Hello, I have a similar problem.

Running on 64 bit Windows with 32 GB RAM 
Game Version: v1.18.3 (Stable)
5/17/2023 2:43:11 PM: Critical error occurred in the following mod: hudclock@2.8.0
Loaded Mods: game@1.18.3, hudclock@2.8.0, creative@1.18.3, survival@1.18.3
System.MissingFieldException: Field not found: 'Vintagestory.API.Common.Entities.Entity.Pos'.
   at hudclock.src.ui.ClaimAreaDialog.RequestClaimInfoOfPlayerPosition(Single gameTick)
   at hudclock.src.ui.ClaimAreaDialog.Build()
   at hudclock.src.busineslogic.ClaimDialogController..ctor(ICoreClientAPI clientApi, ClockModSettingsController clockModSettingsController)
   at hudlock.src.HUDClockMod.PlayerReady(EnumHandling& handling)
   at Vintagestory.Client.NoObf.ClientEventAPI.TriggerIsPlayerReady() in VintagestoryLib\Client\API\ClientEventAPI.cs:line 379
   at Vintagestory.Client.NoObf.GeneralPacketHandler.HandlePlayerData(Packet_Server packet) in VintagestoryLib\Client\Systems\GeneralPacketHandler.cs:line 301
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 747
   at Vintagestory.Client.GuiScreenConnectingToServer.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenConnectingToServer.cs:line 304
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 724
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 627
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 126
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 479
   at OpenTK.GameWindow.DispatchRenderFrame() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 454
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 375
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 5/17/2023 2:36:06 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1869540112483746582, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.18.3.0
P3: 10.0.19044.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\Thomas\AppData\Local\Temp\RDRDCA.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERDDA.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERDEB.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE08.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE29.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 185e2fd3-85c2-4674-aba2-2f5c6872b938
Report Status: 268435456
Hashed bucket: ba6352b6d3a3b814a9f1f0d3b9f00316
Cab Guid: 0 }
--------------
{ TimeGenerated = 4/3/2023 10:33:25 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1632951646305068015, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.17.9.0
P3: 10.0.19044.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\Thomas\AppData\Local\Temp\RDR8465.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8476.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8477.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8494.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER84B4.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 31420711-e1a9-45e2-934e-77acf6306249
Report Status: 268435456
Hashed bucket: 1dc75c8e8b60a89926a968dedd6723ef
Cab Guid: 0 }
--------------
{ TimeGenerated = 11/20/2022 6:07:18 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1804189158136904566, type 4
Event Name: APPCRASH
Response: Non disponible
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.17.9.0
P3: 63501532
P4: nvoglv64.dll
P5: 31.0.15.2698
P6: 63708b78
P7: c0000005
P8: 0000000000e50514
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WEREC35.tmp.WERInternalMetadata.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_4b791b379a40a8c68d866d8bc645deab476cb2f_7fba9aec_b8a6625e-2e24-4fe0-adaa-4e5ac5eaf741

Analysis symbol: 
Rechecking for solution: 0
Report Id: 571469ed-864b-46c7-ab60-6cdf708190db
Report Status: 268435456
Hashed bucket: b1793b8c5ea3c641d909c47af65c7b76
Cab Guid: 0 }

**Comment by Craluminum2413** [comment]
@Inbequer Your crash is related to hud clock mod

**Comment by Craluminum2413** [comment]
Fixed in 1.18.4
```
Fixed: Phantom errors being reported after Cancel / Force Quit button pressed during game startup for example "Mods must not get assets before AssetsLoaded stage" error
```
